src/protocols/ssh/click.c

```diff
@@ -0,0 +1,457 @@
+    /* Write each character to clipboard */
+    
+    while (i <= length){
+
+        int remaining = sizeof(buffer);
+        char* current = buffer;
+
+        for(i = 0; i <= length; i++) {
+            int codepoint = buffer_row->characters[i].value;
+
+            if (codepoint == 0)
+                codepoint = 32;
+
+            if (codepoint == GUAC_CHAR_CONTINUATION)
+                continue;
+
+            int bytes = guac_utf8_write(codepoint, current, remaining);
+
+            if (bytes == 0)
+                break;
+
+            current += bytes;
+            remaining -= bytes;
+        }
+
+        guac_common_clipboard_append(click->term->clipboard, buffer, current - buffer);
+        guac_common_clipboard_send(click->term->clipboard, click->client);
+
+    }    
+
+}
+
+guac_click* guac_click_get_word_border(guac_click* click){
+
+    /* Get select information */
+
+    guac_terminal* term = click->term;
+    
+    int row = click->select_row;
+    int col = click->select_col;
+    int head = click->select_head;
+    int tail = click->select_tail;
+
+    int sentinel = term->display->operations[row * term->display->width + col].character.value;
+    int flag = sentinel;
+            
+    /* Get head */
+    while (((flag > 64 && flag < 91) || (flag > 96 && flag < 123) || (flag > 47 && flag < 58) || (flag == 95)) && (head >= 0 && head <= term->display->width)){
+        flag = term->display->operations[row * term->display->width + head].character.value;
+        head--;
+    }
+    
+
+    /* Reset flag */
+    flag = sentinel;
+
+    /* Get tail */
+    while (((flag > 64 && flag < 91) || (flag > 96 && flag < 123) || (flag > 47 && flag < 58) || (flag == 95)) &&(tail >= 0 && tail <= term->display->width)){
+        flag = term->display->operations[row * term->display->width + tail].character.value;
+        tail++;
+    }
+    
+
+            
+    head += 2;
+    tail -= 2;
+
+    int h = term->display->operations[row * term->display->width + head - 1].character.value;
+    if (head == 1 && ((h > 64 && h < 91) || (h > 96 && h < 123) || (h > 47 && h < 58)  || (h == 95)))
+        head = 0;
+
+    printf("Head: %d\n", head);
+    printf("Tail: %d\n", tail);
+
+    click->select_head = head;
+    click->select_tail = tail;
+
+    return click;
+
+}
+
+guac_click* guac_click_get_blank_border(guac_click* click){
+    
+    /* Get select information */
+    guac_terminal* term = click->term;
+    
+    int row = click->select_row;
+    int col = click->select_col;
+    int head = click->select_head;
+    int tail = click->select_tail;
+
+    int sentinel = term->display->operations[row * term->display->width + col].character.value;
+    int flag = sentinel;
+
+    /* Get head */
+    while ((flag == 0 || flag == 32) && (head >= 0 && head <= term->display->width)){
+        flag = term->display->operations[row * term->display->width + head].character.value;
+        head--;
+    }
+
+    /* Reset flag */
+    flag = sentinel;
```

Comment:
Please rephrase to capture the intent of what's happening here, rather than the literal English equivalent of the relevant statement.